### PR TITLE
OB-185 User API

### PR DESCRIPTION
**Pull request by zesilvapro**
- Adicionei API ...

### CODE REVIEW

**Review by HenriqueMoniz — APPROVED** [review]
Tudo partido mas pode ir para a frente

File: Backend-Roadmap/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.vs/*
```
Comment (tentone) [inline Backend-Roadmap/.gitignore]: Check the gitignore to include all temporary files

File: Backend-Roadmap/Program.cs [hunk 2]
```diff
@@ -13,6 +13,8 @@ public static void Main(string[] args)
             // Add services to the container.
             builder.Services.AddControllers();
 
+ // teste
```
Comment (jl-eqsdigital) [inline Backend-Roadmap/Program.cs]: remove

**Review by tentone — CHANGES_REQUESTED** [review]

File: Backend-Roadmap/.gitignore (hunk 1 above)
Comment (zesilvapro) [reply Backend-Roadmap/.gitignore]: .ignore actualizado

**Comment by zesilvapro** [comment]
- [X] .gitignore 
- [X]  User table CRUD implemented
SwaggerControl

File: Backend-Roadmap/Controllers/UsersController.cs [hunk 3]
```diff
@@ -115,6 +115,66 @@ public ActionResult RegisterUser([FromBody] Users newUser)
                 return StatusCode(500, new { error = "An error occurred while registering the user: " + ex.Message });
             }
         }
+        
+
+        [HttpPut]
+        [Route("UpdateUser/{userId}")]
+        public ActionResult UpdateUser(int userId, Users updatedUser)
```
Comment (jl-eqsdigital) [inline Backend-Roadmap/Controllers/UsersController.cs]: Usually it's
```
public ActionResult UpdateUser(int userId, [FromBody] Users userToUpdate)
```
try naming the variables so we know what they mean

**Review by jl-eqsdigital — APPROVED** [review]

File: Backend-Roadmap/Controllers/UsersController.cs (hunk 3 above)
Comment (jl-eqsdigital) [reply Backend-Roadmap/Controllers/UsersController.cs]: Try also programming asynchronously with async Task<T>  and  await

Comment (zesilvapro) [reply Backend-Roadmap/Controllers/UsersController.cs]: Updated

File: Backend-Roadmap/Program.cs (hunk 2 above)
Comment (zesilvapro) [reply Backend-Roadmap/Program.cs]: deleted

**Review by jl-eqsdigital — APPROVED** [review]

**Review by tentone — APPROVED** [review]

**Pull request merged**